README.md

```diff
@@ -349,21 +349,77 @@ The extended _geonode_config.html template should set the `__GEONODE_CONFIG__.ov
+
+        */
+
+        var cfgPluginOverride = {
+                "map_viewer":[
+                    {
+                        "name":"Search",
+                        "cfg":{
+                            "showCoordinatesSearchOption":false,
+                            "maxResults":20,
+                            "searchOptions":{
+                            "services":[
+                                {
+                                    "type":"wfs",
+                                    "priority":3,
+                                    "displayName":"${properties.propToDisplay}",
+                                    "subTitle":" (a subtitle for the results coming from this service [ can contain expressions like ${properties.propForSubtitle}])",
+                                    "options":{
+                                        "url":"/geoserver/wfs",
+                                        "typeName":"workspace:layer",
+                                        "queriableAttributes":[
+                                        "attribute_to_query"
+                                        ],
+                                        "sortBy":"id",
+                                        "srsName":"EPSG:4326",
+                                        "maxFeatures":20,
+                                        "blacklist":[
+                                        "... an array of strings to exclude from  the final search filter "
+                                        ]
+                                    }
+                                ]
+                            }
+                            }
+                        }
+                    ]
+        }
+
     };
+
+    /*
+
+        This script parse localConfig and cfgPluginOverride
+        if match the plungin name for the geonode section, extend or override it
+
+        if the plugin is new, add it to localconfig
+        */
+        if(localConfig.plugins){
+            for (key in cfgPluginOverride) {
+                if(localConfig.plugins[key]){
+                    for(idx in localConfig.plugins[key]){
+                        for(index in cfgPluginOverride[key]){
+                            if(localConfig.plugins[key][idx].name === cfgPluginOverride[key][index].name){
+                                // extend the plugin configuration or overtide existent property
+                                localConfig.plugins[key][idx] = _.merge(localConfig.plugins[key][idx], cfgPluginOverride[key][index]);
+                            }
+                            // add new obect plugin
+                            if(localConfig.plugins[key].indexOf(cfgPluginOverride[key][index]) == -1){
+                                localConfig.plugins[key].push(cfgPluginOverride[key][index])
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+    return localConfig
```

Comment:
Proposal:

1) Please don't remove the previous example but we could add a list of snippet examples where this should be `update plugin configuration`
2) Add also the example for `enable plugin` (copy from here https://github.com/GeoNode/geonode-mapstore-client/tree/3.3.x#customization-via-configurationstemplates)

Below a refactor to include the logic in a function

---

- update plugin configuration

```html
{% extends 'geonode-mapstore-client/_geonode_config.html' %}
{% block override_local_config %}
<script>
    window.__GEONODE_CONFIG__.overrideLocalConfig = function(localConfig, _) {
        /**
        * this is an example of function used to merge new plugin configuration in the default localConfig
        * if match the plugin name for the GeoNode section, extend or override it
        * if the plugin is new, add it to localConfig
        * Note: you can create your on function or manipulate the localConfig to get your expected final configuration
        * @param {object} config localConfig to update
        * @param {string[]} options.pages array of page keys to target
        * @param {string} options.name name of plugin
        * @param {object} options.cfg new cfg to apply
        */
        function mergePluginConfig(config, options) {
            var pages = options.pages;
            var pluginName = options.name;
            var pluginCfg = options.cfg;
            for (var j = 0; j < pages.length; j++ ) {
                var page = pages[j];
                var plugins = config.plugins[page];
                var merged = false;
                for (var i = 0; i < config.plugins[page].length; i++ ) {
                    var plugin = plugins[i];
                    if (plugin.name === pluginName) {
                        plugin.cfg = _.merge(plugin.cfg, pluginCfg);
                        merged = true;
                        break;
                    }
                }
                if (!merged) {
                    plugins.push({
                        name: pluginName,
                        cfg: pluginCfg
                    })
                }
            }
        }

        mergePluginConfig(localConfig, {
            pages: [ 'map_edit', 'map_view' ], 
            name: 'Search',
            cfg: {
                "searchOptions": {
                    "services": [
                        // { "type": "nominatim", "priority": 5 }, // default service
                        {
                            "type": "wfs",
                            "priority": 3,
                            "displayName": "${properties.propToDisplay}",
                            "subTitle": " (a subtitle for the results coming from this service [ can contain expressions like ${properties.propForSubtitle}])",
                            "options": {
                                "url": "{state('settings') && state('settings').geoserverUrl ? state('settings').geoserverUrl + '/wfs' : '/geoserver/wfs'}",
                                "typeName": "workspace:layer",
                                "queriableAttributes": [
                                    "attribute_to_query"
                                ],
                                "sortBy": "id",
                                "srsName": "EPSG:4326",
                                "maxFeatures": 20,
                                "blacklist": [
                                    "... an array of strings to exclude from  the final search filter "
                                ]
                            }
                        }
                    ]
                }
            }
        });
        return localConfig;
    };
</script>
{% endblock %}
```